frame/benchmarking/src/analysis.rs

```diff
@@ -158,7 +188,7 @@ fn linear_regression(
 		*value -= min;
 	}
 
-	let (intercept, params, errors) = raw_linear_regression(xs, ys, x_vars, false)?;
+	let (intercept, params, errors) = raw_linear_regression(&xs, &ys, x_vars, false)?;
 	Some((intercept + min, params, errors[1..].to_vec()))
```

Comment:
Could you an assertion that the `intercept` is non-negative?

Reply:
Then we can be sure that we dont silently ignore something. Maybe like `intercept.abs() <= .01`.

Reply:
Okay, I've added an `assert`.